akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -290,12 +291,11 @@ private[http] object BodyPartParser {
   }
 
   // TODO: load from config
-  val defaultSettings = Settings(
-    maxHeaderNameLength = 64,
-    maxHeaderValueLength = 8192,
-    maxHeaderCount = 64,
-    illegalHeaderWarnings = true,
-    headerValueCacheLimit = 8,
-    uriParsingMode = Uri.ParsingMode.Relaxed,
-    cookieParsingMode = ParserSettings.CookieParsingMode.RFC6265)
+
+  // TODO: load from config
+  object Settings {
+    def apply(s: ParserSettings): Settings = Settings(s.maxHeaderNameLength, s.maxHeaderValueLength, s.maxHeaderCount, s.illegalHeaderWarnings, s.headerValueCacheLimit("default"), s.uriParsingMode, s.cookieParsingMode)
+  }
+
+  val defaultSettings = Settings(ParserSettings.fromSubConfig(ConfigFactory.load, ConfigFactory.systemProperties()))
```

Comment:
Two goals here:
- remove this val (it's a global static value == bad)
- as config may be sometimes namespaced ("myapp.akka.parsing" and then the config passed into the `ActorSystem` may be `rootConfig.getConfig("myapp")` we must use the config from the ActorSystem, not just `ConfigFactory.load` at some place

To achieve this, we must force the settings to be passed into this class - in places where it is used. And figure out where to parse the config (once).